test/test_x509req.rb

```diff
@@ -151,6 +151,11 @@ def test_eq
     assert_equal false, req1 == req3
   end
 
+  def test_marshal
+    req = issue_csr(0, @dn, @rsa1024, "sha256")
+    assert_equal req, Marshal.load(Marshal.dump(req))
```

Comment:
Most tests are only testing that dump and load are reverse operations.

Would it help to have a test actually testing that it dumps to a string?
If not, you would have green tests by changing the implementation to anything that keeps operations reverse, like both being the "identity" operation e.g.

Reply:
> Would it help to have a test actually testing that it dumps to a string?

Maybe I'm misunderstanding but I don't think we really care about the intermediate format, but more about the fact that the deserialized object is equal to the one we had before serializing it. The current way relies on the equality tests covering that, but I can add some of that to the marshal tests as an extra sanity check.

Reply:
Fair enough :+1: